internal/storage/sql/common/evaluation.go

```diff
@@ -160,3 +160,114 @@ func (s *Store) GetEvaluationDistributions(ctx context.Context, ruleID string) (
 
 	return distributions, nil
 }
+
+func (s *Store) GetEvaluationRollouts(ctx context.Context, flagKey, namespaceKey string) ([]*storage.EvaluationRollout, error) {
+	rows, err := s.db.QueryContext(ctx, `
+		SELECT r.id, r.namespace_key, r."type", r."rank", rp.percentage, rp.value, rss.segment_key, rss.rollout_segment_value, rss.match_type, rss.constraint_type, rss.constraint_property, rss.constraint_operator, rss.constraint_value
+		FROM rollouts r
+		LEFT JOIN rollout_percentages rp ON (r.id = rp.rollout_id)
+		LEFT JOIN (
+			SELECT rs.rollout_id, rs.segment_key, s.match_type, rs.value AS rollout_segment_value, c."type" AS constraint_type, c.property AS constraint_property, c.operator AS constraint_operator, c.value AS constraint_value
+			FROM rollout_segments rs
+			JOIN segments s ON (rs.segment_key = s."key")
+			JOIN constraints c ON (rs.segment_key = c.segment_key)
+		) rss ON (r.id = rss.rollout_id)
+		WHERE r.flag_key = $1 AND r.namespace_key = $2
+		ORDER BY r."rank" ASC
```

Comment:
this is a monster query, im wondering if it would be simpler and potentially faster because all of the joins to do:

1. get all rollouts for flag from `rollouts` table by flagKey, sorted by `rank`, this will give you a slice of `[]rollouts` that are naturally ordered
2. create a `map[rolloutID] => rollout` as an index for each type (percentage & variant) 
3. query for all `percentage_rollouts` by ID, using an `IN()` query with their IDs from step 2
4. query for all `segment_rollouts` by ID, using an `IN()` query with their IDs from step 2
5. now sort the final results looping through the id's in order from step 1, and built up the final result set using the actual `percentage_rollout` or `segment_rollout`. 

I guess what I'm saying is I wonder which approach will be faster, implementing it all in SQL or in Go?
Perhaps we should create some benchmarks and try both approaches? We could create another issue for this to do before it goes 'live'. WDYT?

Reply:
@markphelps Benchmarks sound good. I meant to call this query out initially, I should have followed up with that.

Reply:
To further this. I think we should break this PR in two like we originally planned.
Focus on delivering the server definition first. Just leave the storage requirement as an interface with no implementation.

Then implement the storage as a follow up PR as there is a lot to digest here.
FLI-459 is the server issue
FLI-454 is the storage implementation

Reply:
@GeorgeMac Agreed. I'll move that to another PR.